ot/lowrank.py

```diff
@@ -0,0 +1,502 @@
+"""
+Low rank OT solvers
+"""
+
+# Author: Laurène David <laurene.david@ip-paris.fr>
+#
+# License: MIT License
+
+
+import warnings
+from .utils import unif, dist, get_lowrank_lazytensor
+from .backend import get_backend
+from .bregman import sinkhorn
+from sklearn.cluster import KMeans
+
+# ADD FUNCTION FOR LOW RANK INIT [WIP]
+
+
+def _init_lr_sinkhorn(X_s, X_t, a, b, rank, init, reg_init=None, random_state=None, nx=None):
+    """
+    Implementation of different initialization strategies for the low rank sinkhorn solver (Q ,R, g).
+    This function is specific to lowrank_sinkhorn.
+
+    Parameters
+    ----------
+    X_s : array-like, shape (n_samples_a, dim)
+        samples in the source domain
+    X_t : array-like, shape (n_samples_b, dim)
+        samples in the target domain
+    a : array-like, shape (n_samples_a,)
+        samples weights in the source domain
+    b : array-like, shape (n_samples_b,)
+        samples weights in the target domain
+    rank : int
+        Nonnegative rank of the OT plan.
+    init : str
+        Initialization strategy for Q, R and g. 'random', 'trivial' or 'kmeans'
+    reg_init : float, optional. Default is None. (>0)
+        Regularization term for a 'kmeans' init. If None, 1 is considered.
+    random_state : default None
+        Random state for a "random" or 'kmeans' init strategy
+    nx : default None
+        POT backend
+
+
+    Returns
+    ---------
+    Q : array-like, shape (n_samples_a, r)
+        Init for the first low-rank matrix decomposition of the OT plan (Q)
+    R: array-like, shape (n_samples_b, r)
+        Init for the second low-rank matrix decomposition of the OT plan (R)
+    g : array-like, shape (r, )
+        Init for the weight vector of the low-rank decomposition of the OT plan (g)
+
+
+    References
+    -----------
+    .. [65] Scetbon, M., Cuturi, M., & Peyré, G. (2021).
+        "Low-rank Sinkhorn factorization". In International Conference on Machine Learning.
+
+    """
+
+    if nx is None:
+        nx = get_backend(X_s, X_t, a, b)
+
+    if reg_init is None:
+        reg_init = 0.1
+
+    if random_state is None:
+        random_state = 49
+
+    ns = X_s.shape[0]
+    nt = X_t.shape[0]
+    r = rank
+
+    if init == "random":
+        nx.seed(seed=random_state)
+
+        # Init g
+        g = nx.abs(nx.randn(r, type_as=X_s)) + 1
+        g = g / nx.sum(g)
+
+        # Init Q
+        Q = nx.abs(nx.randn(ns, r, type_as=X_s)) + 1
+        Q = (Q.T * (a / nx.sum(Q, axis=1))).T
+
+        # Init R
+        R = nx.abs(nx.randn(nt, rank, type_as=X_s)) + 1
+        R = (R.T * (b / nx.sum(R, axis=1))).T
+
+    if init == "trivial":
```

Comment:
This initialization strategy was called "trivial" by the researchers.
I don't really know how to name it otherwise.

I can also remove this option for init, since there is also a "random" and "kmeans" option.

Reply:
Thank you for your suggestion. I'll change it to "deterministic".